### PR TITLE
Slider tooltip and letterboxing/dim changes 

### PR DESCRIPTION
**Pull request by jorolf**
Depends on https://github.com/ppy/osu-framework/pull/649

### CODE REVIEW

File: osu.Game/Graphics/UserInterface/OsuSliderBar.cs [hunk 1]
```diff
@@ -23,16 +25,20 @@ public class OsuSliderBar<T> : SliderBar<T>, IHasTooltip where T : struct
...
+                if (TooltipTextFunc != null) return TooltipTextFunc(Current);
+
                 var bindableDouble = CurrentNumber as BindableNumber<double>;
                 if (bindableDouble != null)
                 {
-                    if (bindableDouble.MaxValue == 1 && bindableDouble.MinValue == 0)
+                    if (bindableDouble.MaxValue == 1 && (bindableDouble.MinValue == 0 || bindableDouble.MinValue == -1))
                         return bindableDouble.Value.ToString(@"P0");
-                    return bindableDouble.Value.ToString(@"n1");
+                    return bindableDouble.Value.ToString(@"n1", CultureInfo.InvariantCulture);
```
Comment (peppy) [inline osu.Game/Graphics/UserInterface/OsuSliderBar.cs]: pretty sure we want variant culture

Comment (huoyaoyuan) [inline osu.Game/Graphics/UserInterface/OsuSliderBar.cs]: seems locale engine can provide the `CultureInfo` for `ToString()`?

File: osu.Game/Graphics/UserInterface/OsuSliderBar.cs [hunk 2]
```diff
@@ -23,16 +25,20 @@ public class OsuSliderBar<T> : SliderBar<T>, IHasTooltip where T : struct
         private readonly Box leftBox;
         private readonly Box rightBox;
 
+        public Func<T, string> TooltipTextFunc { get; set; }
```
Comment (peppy) [inline osu.Game/Graphics/UserInterface/OsuSliderBar.cs]: why not just make `TooltipText` `virtual` and make classes for special cases? i think that's a lot more elegant than this.

**Comment by peppy** [comment]
can you rebase your stuff?
9/12 of your commits are merges of master...

**Comment by Jorolf** [comment]
_rebased_

File: osu.Game/Overlays/Options/Sections/SkinSection.cs [hunk 3]
```diff
@@ -59,15 +58,15 @@ private void load(OsuConfigManager config)
...
                 },
-                new OptionSlider<double>
+                new OptionSlider<double,SizeSlider>
                 {
                     LabelText = "Menu cursor size",
-                    Bindable = (BindableDouble)config.GetBindable<double>(OsuConfig.MenuCursorSize)
+                    Bindable = config.GetBindable<double>(OsuConfig.MenuCursorSize)
                 },
-                new OptionSlider<double>
+                new OptionSlider<double,SizeSlider>
```
Comment (peppy) [inline osu.Game/Overlays/Options/Sections/SkinSection.cs]: space after comma

**Review by peppy — CHANGES_REQUESTED** [review]

**Comment by peppy** [comment]
This can't be merged until https://github.com/ppy/osu-framework/pull/659 is merged. Everyone who has an existing setting for dim will have it set incorrectly as there is no consideration given to migration.

**Review by peppy — APPROVED** [review]

**Pull request merged**